diff --git a/train_emotion_classifier.py b/train_emotion_classifier.py
index 93433cd..85261b9 100644
--- a/train_emotion_classifier.py
+++ b/train_emotion_classifier.py
@@ -8,12 +8,21 @@ from PIL import Image
 import keras
 import subprocess
 import os
+from keras.layers import Conv2D, MaxPooling2D, Dropout, Dense, Flatten
 
 import wandb
 from wandb.keras import WandbCallback
 
 run = wandb.init()
 config = run.config
+config.first_layer_convs = 32
+config.first_layer_conv_width = 3
+config.first_layer_conv_height = 3
+config.dropout = 0.2
+config.dense_layer_size = 128
+config.img_width = 28
+config.img_height = 28
+
 
 config.batch_size = 32
 config.num_epochs = 20
@@ -53,12 +62,24 @@ train_faces /= 255.
 val_faces /= 255.
 
 model = Sequential()
+model.add(Conv2D(32,
+                 (config.first_layer_conv_width, config.first_layer_conv_height),
+                 input_shape=(48, 48, 1),
+                 activation='relu'))
+model.add(MaxPooling2D(pool_size=(4, 4)))
+model.add(Dropout(0.5))
+model.add(Conv2D(64,(4,4),
+                 activation='relu'))
+model.add(MaxPooling2D(pool_size=(2, 2)))
+model.add(Dropout(0.5))
 model.add(Flatten(input_shape=input_shape))
+model.add(Dense(config.dense_layer_size, activation='relu'))
 model.add(Dense(num_classes, activation="softmax"))
 
+
 model.compile(optimizer='adam', loss='categorical_crossentropy',
 metrics=['accuracy'])
-
+config.total_params = model.count_params()
 model.fit(train_faces, train_emotions, batch_size=config.batch_size,
         epochs=config.num_epochs, verbose=1, callbacks=[
             WandbCallback(data_type="image", labels=["Angry", "Disgust", "Fear", "Happy", "Sad", "Surprise", "Neutral"])
